core/src/main/scala/zio/jdbc/JdbcEncoder.scala

```diff
@@ -16,70 +16,123 @@
 package zio.jdbc
 
 import zio.Chunk
+import zio.jdbc.SqlFragment.{ Segment, Setter }
 import zio.schema.{ Schema, StandardType }
 
+import java.sql.PreparedStatement
+
 /**
  * A type class that describes the ability to convert a value of type `A` into
  * a fragment of SQL. This is useful for forming SQL insert statements.
  */
-trait JdbcEncoder[-A] {
+trait JdbcEncoder[A] {
   def encode(value: A): SqlFragment
+  val setter: Option[Setter[A]]
```

Comment:
First, I would inline `Setter`, as there is no need for a separate type class.

Second, why make it `Option`? Are there some cases that have no well-defined implementation?

Reply:
Well, the one from issue does not:
```
  implicit val byteArrayjdbcEncoder: JdbcEncoder[Array[Byte]] =
    value => s"FROM_BASE64('${encoder.encodeToString(value)}')"
```

I mean - setter can be set as required, but encoder won't be so simple.

Maybe it's better - after all these kind of encoders are open to SQL injection attacks.

Reply:
So after seeing support for iterable I had an epiphany how it may work - I hope API is now what you wanted.